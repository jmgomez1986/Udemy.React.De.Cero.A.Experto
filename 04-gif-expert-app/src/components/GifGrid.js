import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

	const [images, setImages] = useState([]);

	useEffect(() => {
		getGifs();
	}, []); // Al mandar la lista  de dependencias vacia, se ejecta solo en la primera renderizacion

	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=5Aur2asApFfGBynHWjHEVbAzIUjLAKK5`
		const response = await fetch(url);
		const {data} = await response.json();

		const gifs = data.map( (img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url
			}
		});

		setImages(gifs);
	}

	return (
		<>

			<h3>{category}</h3>

			<div className="card-grid">
				
				{
					images.map( (img) => 
						<GifGridItem
							key={img.id}
							// img={img}
							{...img}
						/>
					)
				}

			</div>
			
		</>
	)
}
