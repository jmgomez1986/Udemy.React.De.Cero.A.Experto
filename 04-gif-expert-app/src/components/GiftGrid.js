import React from 'react'

export const GiftGrid = ({category}) => {

	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=5Aur2asApFfGBynHWjHEVbAzIUjLAKK5`
		const response = await fetch(url);
		const {data} = await response.json();

		const gifs = data.map( (img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url
			}
		})

		console.log(gifs);
	}

	getGifs();

	return (
		<div>
			<h3>{category}</h3>
		</div>
	)
}
