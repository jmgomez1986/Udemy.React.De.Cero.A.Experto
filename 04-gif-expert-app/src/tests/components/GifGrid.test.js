import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componente <GifGrid />', () => {
	const category = 'goku';
	
	test('debe de mostrarse el componente <GifGrid /> correctamente ', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wraper = shallow(<GifGrid category={category}/>);

		expect(wraper).toMatchSnapshot();
	});

	test('debe de mostraar items cuando se cargan imagenes con useFetchGifs ', () => {
		const gifs = [
			{
			id: 'ABC',
			url: 'https://localhost/cualquier/cosa.jpg',
			title: 'Cualquier cosa'
		},
		{
			id: '123',
			url: 'https://localhost/cualquier/cosa.jpg',
			title: 'Cualquier cosa'
		}
	];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wraper = shallow(<GifGrid category={category}/>);

		// expect(wraper).toMatchSnapshot();

		// El parrafo de Loading no debe existir
		expect(wraper.find('p').exists()).toBe(false);
		// Comprobar que este el componente <GifGridItem />
		expect(wraper.find('GifGridItem').length).toBe(gifs.length);

	});	
	
});
