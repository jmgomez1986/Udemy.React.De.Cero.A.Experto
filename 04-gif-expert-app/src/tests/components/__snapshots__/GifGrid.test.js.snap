// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid /> debe de mostraar items cuando se cargan imagenes con useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas del componente <GifGrid /> debe de mostrarse el componente <GifGrid /> correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
